engine/access/rest/websockets/data_providers/base_provider.go

```diff
@@ -0,0 +1,64 @@
+package data_providers
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+
+	"github.com/onflow/flow-go/engine/access/subscription"
+)
+
+// BaseDataProvider defines the basic interface for a data provider. It provides methods
+// for retrieving the provider's unique ID, topic, and a method to close the provider.
+type BaseDataProvider interface {
+	// ID returns the unique identifier of the data provider.
+	ID() uuid.UUID
+	// Topic returns the topic associated with the data provider.
+	Topic() string
+	// Close terminates the data provider.
+	Close() error
+}
+
+var _ BaseDataProvider = (*BaseDataProviderImpl)(nil)
```

Comment:
if this is the main reason to separate `BaseDataProvider`, I think you can skip the check since it should be done explicitly for all of the main providers

Reply:
[ff3851b](https://github.com/onflow/flow-go/pull/6636/commits/ff3851b346f1c9a5e638dd1f54ae1a5506c8e0bd)